DiagManager/CustomDiagnostics/AlwaysOn Basic Info/AlwaysOnDiagScript.sql

```diff
@@ -42,37 +58,55 @@ PRINT '==================================='
 PRINT 'Availability Group Listeners and IP'
 PRINT '==================================='
 PRINT ''
+PRINT '-- AG_hadr_ag_listeners --'
 --First the listeners, one line per listener instead of the previous multi-line per IP.
 -- IPs will be broken out in the next query.
-SELECT agl.dns_name AS [Listener_Name], ag.name AS [AG_Name]
-		, agl.*
-FROM sys.availability_group_listeners agl
-	inner join sys.availability_groups ag
-		on agl.group_id = ag.group_id
+DECLARE @sql_major_version INT, @sql_major_build INT, @sql NVARCHAR(max)
 
---IP information which isn't fully returned via the query above.
-SELECT agl.dns_name AS [Listener_Name], aglip.*
-FROM sys.availability_group_listener_ip_addresses aglip
-	inner join sys.availability_group_listeners agl
-		on aglip.listener_id = agl.listener_id
+SELECT @sql_major_version = (CAST(PARSENAME(CAST(SERVERPROPERTY('ProductVersion') AS varchar(20)), 4) AS INT))
+       
+
+SET @sql ='SELECT agl.dns_name AS [Listener_Name], ag.name AS [AG_Name] ,agl.group_id,listener_id,dns_name,port,is_conformant,ip_configuration_string_from_cluster'
+
+IF (@sql_major_version >=14) --this exists SQL 2017 and above
+	BEGIN
+	  SET @sql = @sql + ',is_distributed_network_name'
+	END
```

Comment:
I ran the script on a SQL 2019 and got this error:
```
-- AG_hadr_ag_listeners --
Msg 207, Level 16, State 1, Server NODE1LABCSS\SQL2019, Line 1
Invalid column name 'is_distributed_network_name'.

Then ran `select * from sys.availability_group_listeners` and don't see this column. The suggestion is to remove this logic or double-check it and make sure it applies to this DMV

```
```suggestion

```

Reply:
Thank you @PiJoCoder , I see this is not there on the RTM and  added later in the CUs and not failed for me since I have those CUs on my test box. I will modify the logic to check the builds as well per major version. 
https://docs.microsoft.com/en-us/azure/azure-sql/virtual-machines/windows/availability-group-distributed-network-name-dnn-listener-configure?view=azuresql

Reply:
Oh, I see, it is the SYS.AVAILABILITY_GROUP_LISTENERS DMV that has this.  I do have it on my SQL 2019 build, but was querying the sys.availability_groups instead. And since the documentation does not include the column, I didn't know which one had it. Can you fully-qualify the column to help with this discovery - ag1.is_distributed_network_name? And perhaps for consistency these columns too: listener_id,dns_name,port,is_conformant,ip_configuration_string_from_cluster
Thank you

Reply:
@PiJoCoder , thank you , I modified this to make names full-qualified for those columns and added the additional build check logic for is_distributed_network_name column. 
--this column exists  SQL 2019 CU8 ,SQL 2017 CU25,SQL 2016 SP3 and above

Reply:
--this column exists SQL 2019 CU8 ,SQL 2017 CU25,SQL 2016 SP3 and above
Thanks for calling these out. Based on these, submitted an update request on 
https://docs.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-availability-group-listeners-transact-sql?view=sql-server-ver16